Dockerfile

```diff
@@ -0,0 +1,64 @@
+FROM ubuntu:bionic
+
+# Set up dependencies
+ENV GCC_VER "10"
+ENV LLVM_VER "11"
+ENV CC gcc-${GCC_VER}
+ENV CXX g++-${GCC_VER}
+
+# Add ability to install PPAs
+RUN apt-get update -qq
+RUN apt-get -y install software-properties-common wget git
+
+# Add repo for gcc-10
+RUN add-apt-repository -y ppa:ubuntu-toolchain-r/test
+# Add repo for clang-11
+RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+RUN add-apt-repository -y "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-${LLVM_VER} main"
+# Add repo for current cmake/ccmake, from Kitware
+RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null \
+    | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
+RUN apt-add-repository 'deb https://apt.kitware.com/ubuntu/ bionic main'
+RUN apt-get update -qq
+
+# Install all the packages
+RUN apt-get install -y python3 python3-pip doxygen graphviz ccache cppcheck \
+    gcc-${GCC_VER} g++-${GCC_VER} \
+    clang-${LLVM_VER} lldb-${LLVM_VER} lld-${LLVM_VER} clangd-${LLVM_VER} \
+    llvm-${LLVM_VER}-dev libclang-${LLVM_VER}-dev clang-tidy-${LLVM_VER} \
+    cmake cmake-curses-gui \
+    neovim emacs
```

Comment:
Maybe add `nano`? Do people use `nano`? 

Does anyone care that it's `neovim` and not `vi`/`vim`?